File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.py
def solution(cards1, cards2, goal):
    result = []
    idx1 = idx2 = 0
    
    for i in range(len(goal)):
        if idx1 < len(cards1) and goal[i] == cards1[idx1]:
            result.append(goal[i])
            idx1 += 1
        elif idx2 < len(cards2) and goal[i] == cards2[idx2]:
            result.append(goal[i])
            idx2 += 1
        else:
            break

    return "Yes" if goal == result else "No"

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    var answer = [];
    let idx1 = 0, idx2 = 0;
    for(let i = 0 ; i < goal.length ; i++){
        if(idx1 < cards1.length && goal[i] === cards1[idx1]) answer.push(cards1[idx1++]);
        else if(idx2 < cards2.length && goal[i] === cards2[idx2]) answer.push(cards2[idx2++]);
        else break;
    }
    return JSON.stringify(goal) === JSON.stringify(answer) ? "Yes" : "No";
}